cyclonedx_py/client.py

```diff
@@ -111,7 +111,7 @@ def get_output(self) -> BaseOutput:
                   '',
                   sep='\n', file=sys.stderr)
 
-        bom = Bom(components=filter(self._component_filter, parser.get_components()))
+        bom = Bom(components=filter(self._component_filter, parser.get_components()), metadata=parser._metadata)
```

Comment:
:x: do not access protected/private properties! 
use the public getters/properties!

Reply:
The current implementation of the `BaseParser` does not provide a get_metadata. Hence the direct access to the properties

Reply:
feel free to create one.

Reply:
would you prefer to extend it in the python-lib or shall I write a ExtendedBaseParser in this project?